pkg/webui/console/views/device-add/repository/connect.js

```diff
@@ -27,22 +27,28 @@ import {
 import { checkFromState, mayEditApplicationDeviceKeys } from '@console/lib/feature-checks'
 
 import { getTemplate } from '@console/store/actions/device-repository'
+import { getApplicationDevEUICount } from '@console/store/actions/applications'
 
 import {
   selectDeviceTemplate,
   selectDeviceTemplateFetching,
   selectDeviceTemplateError,
 } from '@console/store/selectors/device-repository'
-import { selectSelectedApplicationId } from '@console/store/selectors/applications'
+import {
+  selectApplicationDevEUICount,
+  selectSelectedApplicationId,
+} from '@console/store/selectors/applications'
 import { selectJoinEUIPrefixes } from '@console/store/selectors/join-server'
 
 const mapStateToProps = state => ({
   appId: selectSelectedApplicationId(state),
+  applicationDevEUICounter: selectApplicationDevEUICount(state),
   prefixes: selectJoinEUIPrefixes(state),
   template: selectDeviceTemplate(state),
   templateFetching: selectDeviceTemplateFetching(state),
   templateError: selectDeviceTemplateError(state),
   createDevice: (appId, device) => api.device.create(appId, device),
+  generateDevEUI: appId => api.application.generateDevEUI(appId),
```

Comment:
Note that this will likely cause a rerender whenever the state is recalculated (which will happen a lot when the app received many events) since the value is always reassigned. It would be better to set this via `setCallback()` within the component.